File: shuju/binary_search.c
#include<stdio.h>

int binary_search(int key,int a[],int n)
{
	int low,high,mid,count=0,flag=0;
	low=0;
	high=n-1;
	while(low<high)
	{
		count++;
		mid=(low+high)/2;
		if(key<a[mid])
		{
			high=mid-1;
		}
		else if(key>a[mid])
		{
			low=mid+1;
		}
		else if(key==a[mid])
		{
			flag=1;
			printf("查找成功！\n查找%d次！a[%d]=%d\n",count,mid,key);
			break;
		}
	}
	if(!flag)
	{
		printf("查找元素不存在！\n");
	}
	return 0;
}

int main()
{
	int i,key,a[100],n;
	printf("请输入数组长度:");
	scanf("%d",&n);
	printf("请输入数组元素：");
	for(i=0;i<n;i++)
	{
		scanf("%d",&a[i]);
	}
	printf("请输入你需要查找的元素：");
	scanf("%d",&key);
	binary_search(key,a,n);

	return 0;
}

File: shuju/binary_search.py
# coding: utf-8

def binary_search(key,a,n):
	low=0
	high=n-1
	count=0
	flag=False
	while low<high:
		count+=1
		mid=int((low+high)/2)
		if key>a[mid]:
			low=mid+1
		elif key<a[mid]:
			high=mid-1
		elif key==a[mid]:
			print("查找次数：%d" % count)
			flag=True
			break
	if not flag:
		print("查找元素不存在！")

if __name__=="__main__":
	binary_search(2000,range(3000),3000)
